src/shapes.cpp

```diff
@@ -564,6 +556,70 @@ void Mesh::computeVertexNormals()
   }
 }
 
+void Mesh::computeWeightedVertexNormals()
+{
+  if (!triangle_normals)
+    computeTriangleNormals();
+  if (vertex_count && !vertex_normals)
+    vertex_normals = new double[vertex_count * 3];
+  EigenSTL::vector_Vector3d avg_normals(vertex_count, Eigen::Vector3d(0, 0, 0));
```

Comment:
Consider using an `Eigen::Map<Eigen::MatrixXd>` to `vertex_normals` (a vertex_count x 3 matrix) to avoid allocating an extra array (and final data copying).

Reply:
I like this suggestion. To match data storage order, either we need a 3xN or Nx3 (with Row major). Not sure if there is much difference to the two, so I'm going to go with ``Eigen::Map<Eigen::Matrix<double, 3, Eigen::Dynamic>>``.